product_sequence/models/product_category.py

```diff
@@ -39,20 +41,65 @@ def _prepare_ir_sequence(self, prefix):
     @api.multi
     def write(self, vals):
         prefix = vals.get("code_prefix")
-        if prefix:
-            for rec in self:
-                if rec.sequence_id:
-                    rec.sudo().sequence_id.prefix = prefix
-                else:
+        for rec in self:
+            if vals.get("parent_id"):
+                categories = self.env["product.category"].search(
+                    [("id", "=", vals["parent_id"])], limit=1
+                )
+                for category in categories:
```

Comment:
Como has puesto un limit en el search no es necesario que hagas un for. Bastaría con un if para comprobar que ha encontrado algo

Reply:
Entendido, lo he modificado!